### PR TITLE
fix(x86): preserve regalloc pressure for reload recipes

### PR DESCRIPTION
**Pull request by tignear**
## Summary

- apply the native regalloc reload-recipe pressure fix from #712 to `master`
- retain hazardous home-transfer snapshots
- cover home-transfer cache and register-pressure edge cases

## Validation

- `cargo +1.98.0 fmt --all -- --check`
- `cargo +1.98.0 test -p celox-backend-x86` (499 passed)
- `cargo +1.98.0 clippy -p celox-backend-x86 --all-targets -- -D warnings`

### CODE REVIEW

File: crates/celox-backend-x86/src/backend/native/regalloc/reconstruct.rs [hunk 1]
```diff
@@ -1180,6 +1162,75 @@ fn prepare_recipe(
...
+            continue;
+        };
+        let ResolvedBase::State(state) = base else {
+            continue;
+        };
+        let Some(previously_loaded) = transfers[..later]
+            .iter()
+            .rposition(|(previous_base, _)| previous_base.as_ref() == Some(base))
+        else {
+            continue;
```
[the remark this answers is not in the document]
Comment (tignear) [reply crates/celox-backend-x86/src/backend/native/regalloc/reconstruct.rs]: Fixed in d346ccb92. Hazard detection now includes a later base whose first use follows an overlapping transfer store, and reconstruction snapshots that base before serial home transfers. Added `overlapping_transfer_store_snapshots_a_later_first_state_base`.

File: crates/celox-backend-x86/src/backend/native/regalloc/reconstruct.rs [hunk 2]
```diff
@@ -1158,17 +1136,21 @@ fn prepare_recipe(
         cache.bases.insert(expected.base.clone(), base);
         base
```
[the remark this answers is not in the document]
Comment (tignear) [reply crates/celox-backend-x86/src/backend/native/regalloc/reconstruct.rs]: Fixed in d346ccb92. Spill planning now reserves the transfer result slot plus a conservative slot for every potentially retained home-transfer base, parking enough surviving residents before reconstruction. Added a pressure-reservation regression test.

File: crates/celox-backend-x86/src/backend/native/regalloc/reconstruct.rs [hunk 3]
```diff
@@ -1158,17 +1136,21 @@ fn prepare_recipe(
...
+    if instructions.is_empty() {
+        // A cached base-only recipe still needs its own SSA definition. The
+        // copy preserves the insertion-point snapshot while giving this
+        // transfer a unique final value for reconstruction and verification.
+        let destination = alloc_fresh(func, logical_for_vreg, logical)?;
+        instructions.push(MInst::Mov {
+            dst: destination,
+            src: current,
+        });
+        expected.steps.push(PureStep::Copy64);
```
[the remark this answers is not in the document]
Comment (tignear) [reply crates/celox-backend-x86/src/backend/native/regalloc/reconstruct.rs]: Fixed in d346ccb92. A reload is now recorded as `MaterializedStateReload` only when its prepared recipe is exactly the direct SimState load for that home. Cached copy-based results go through the generic recipe verifier instead. The cached base-only test now covers both classifications.

File: crates/celox-backend-x86/src/backend/native/regalloc/reconstruct.rs [hunk 4]
```diff
@@ -1180,6 +1199,69 @@ fn prepare_recipe(
...
+    operations: &[PlannedEdgeOp],
+    recipes: &[Option<ResolvedRecipe>],
+    state_homes: &std::collections::BTreeMap<SpillHome, crate::native::mir::PackedStateHome>,
+) -> HashSet<ResolvedBase> {
+    let mut cacheable = HashSet::default();
+    let transfers = operations
+        .iter()
+        .enumerate()
+        .filter_map(|(index, operation)| {
+            let PlannedEdgeOp::Reload { destination, .. } = operation else {
```
[the remark this answers is not in the document]
Comment (tignear) [reply crates/celox-backend-x86/src/backend/native/regalloc/reconstruct.rs]: Fixed in b74f51413. The register-hoisting approach has been removed. Hazard analysis now considers every allocator-managed state-home spill at the edge insertion point, including leading resident spills, and falls overlapping homes back to stack storage before reconstruction.

File: crates/celox-backend-x86/src/backend/native/regalloc/reconstruct.rs [hunk 5]
```diff
@@ -285,18 +279,59 @@ pub(super) fn reconstruct(
...
+            bases: HashMap::default(),
+        };
+        for (operation, recipe) in operations.iter().zip(&recipes) {
+            let PlannedEdgeOp::Reload { source, .. } = *operation else {
+                continue;
+            };
+            let Some(recipe) = recipe else {
+                continue;
+            };
+            if !recipe_cache.cacheable_bases.contains(&recipe.base)
```
[the remark this answers is not in the document]
Comment (tignear) [reply crates/celox-backend-x86/src/backend/native/regalloc/reconstruct.rs]: Fixed in b74f51413 by removing simultaneous base hoisting entirely. Any selected state home whose edge store overlaps a selected reload recipe is retired (with its overlap component) to a stack home, so transfer count no longer creates retained base live ranges. Added hazard-detection and verified stack-fallback regressions.
